lib/adb-device.js

```diff
@@ -214,6 +261,43 @@ class ADBDevice {
     await this.sendFile(source, destination, localId, remoteId);
   }
 
+  // runs ADB push to push a file to the device
+  async list (remotePath) {
+    let syncBuf = new Buffer("sync:.");
+    // open a SYNC stream on the device for pushing the file
+    let packet = await this.sendAndOkay(ADB_COMMANDS.CMD_OPEN
+                                       , 12345
+                                       , 0
+                                       , syncBuf);
+    // this is our local id, the devices remote id
+    let localId = packet.arg2;
+    // this is our remote id, the devices local id
+    let remoteId = packet.arg1;
+    packet = await this.stat(remotePath, localId, remoteId);
+    let mode = packet.data.readUInt32LE(4);
+    console.log(mode, packet.data.readUInt32LE(0), packet.data.readUInt32LE(8), packet.data.readUInt32LE(12));
```

Comment:
minor issue: we should either use appium-logger as with other packages, or omit. we try to never use console.log directly.